### PR TITLE
Add data augmentation

### PR DESCRIPTION
**Pull request by kaminow**
Add functionality for on-the-fly data augmentation during the training stage.

### CODE REVIEW

**Comment by hmacdope** [comment]
@kaminow is this ready for review or still WIP?

**Comment by kaminow** [comment]
still doing some validation here but should be ready soon

**Comment by hmacdope** [comment]
@kaminow per discussion we might remove Bfactor based jittering in this PR and revisit down the track.

**Comment by kaminow** [comment]
@hmacdope I've gotten rid of the B-factor based jittering for now until I sort the math out so that we can get this initial stuff merged in. the current implementation of B-factor jittering is sitting in https://github.com/choderalab/asapdiscovery/tree/add-b-factor-jitter, and I'll keep working in that branch

**Comment by kaminow** [comment]
which is to say this PR is now ready for review pls

File: asapdiscovery-ml/asapdiscovery/ml/cli.py [hunk 1]
```diff
@@ -248,6 +249,10 @@ def build_gat(
         "loss_type": loss_type,
         "semiquant_fill": semiquant_fill,
     }
+    data_aug_configs = [
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/cli.py:252]: Can you give an example of the string you are splitting here?

File: asapdiscovery-ml/asapdiscovery/ml/cli_args.py [hunk 2]
```diff
@@ -1250,6 +1259,21 @@ def device(func):
     return click.option("--device", type=torch.device, help="Device to train on.")(func)
 
 
+def data_aug(func):
+    return click.option(
+        "--data-aug",
+        type=str,
+        multiple=True,
+        help=(
+            "Specifications for data augmentations to do. Multiple can be passed, and "
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/cli_args.py:1268]: GIve example of one that you can use.

**Review by hmacdope — APPROVED** [review]
Looks good over all, just one tiny quibble, but won't block

**Comment by kaminow** [comment]
yep good call, thanks!

**Pull request merged**